### PR TITLE
Build the npm package in this repo

### LINKED ISSUE #4212: Produce NPM package in this repo
Rather than manually producing the package in the bower-signalr repo, we should produce the NPM package in this repo and include a proper README.

### PR DESCRIPTION
**Pull request by anurse**
fixes #4212 by putting the npm package build in this repo

TODO:
* [x] ~~Verify CI build output.~~ - The package is confirmed in the CI output

### CODE REVIEW

File: src/Microsoft.AspNet.SignalR.JS/LICENSE.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+Copyright (c) Microsoft Open Technologies, Inc.  All rights reserved.
```
Comment (anurse) [inline src/Microsoft.AspNet.SignalR.JS/LICENSE.md]: Taken as-is from https://github.com/SignalR/bower-signalr

File: src/Microsoft.AspNet.SignalR.JS/package.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
+{
```
Comment (anurse) [inline src/Microsoft.AspNet.SignalR.JS/package.json]: Taken from https://github.com/SignalR/bower-signalr

File: src/Microsoft.AspNet.SignalR.JS/Microsoft.AspNet.SignalR.JS.csproj [hunk 3]
```diff
@@ -21,16 +21,30 @@
...
+    <JSCompile Include="LICENSE.md" />
+    <JSCompile Include="README.md" />
   </ItemGroup>
 
   <ItemGroup>
-    <GeneratedJavaScript Include="$(OutDir)jquery.signalR-$(PackageVersion).js" />
-    <GeneratedJavaScript Include="$(OutDir)jquery.signalR-$(PackageVersion).min.js" />
+    <GeneratedJavaScript Include="$(OutputPath)jquery.signalR-$(PackageVersion).js" />
+    <GeneratedJavaScript Include="$(OutputPath)jquery.signalR-$(PackageVersion).min.js" />
+    <GeneratedJavaScript Include="$(OutputPath)jquery.signalR.min.js" />
```
Comment (BrennanConroy) [inline src/Microsoft.AspNet.SignalR.JS/Microsoft.AspNet.SignalR.JS.csproj]: We're now generating 2 more .js files?

Comment (BrennanConroy) [reply src/Microsoft.AspNet.SignalR.JS/Microsoft.AspNet.SignalR.JS.csproj]: Are the versions embedded in them?

Comment (anurse) [reply src/Microsoft.AspNet.SignalR.JS/Microsoft.AspNet.SignalR.JS.csproj]: a) we were already generating them, this just adds them to the Outputs for MSBuild to detect that they need to be rebuilt

b) these are for the tests and NPM package, both of which expect a versionless file path

c) yes, the versions are embedded in all of the JS files (even the ones with the version number in the file name)

**Pull request merged**